packages/data-context/src/sources/MigrationDataSource.ts

```diff
@@ -0,0 +1,75 @@
+import path from 'path'
+import type { DataContext } from '..'
+
+type ConfigOptions = {
+  global: Record<string, unknown>
+  e2e: Record<string, unknown>
+  component: Record<string, unknown>
+}
+
+export class MigrationDataSource {
+  constructor (private ctx: DataContext) { }
+
+  async getConfig () {
+    const config = await this.parseCypressConfig()
+
+    return JSON.stringify(config, null, 2)
+  }
+
+  private parseCypressConfig (): Promise<Cypress.ConfigOptions> {
+    const cfgPath = path.join(this.ctx.lifecycleManager?.projectRoot, 'cypress.json')
+
+    return this.ctx.file.readJsonFile(cfgPath)
+  }
+
+  async createConfigString () {
+    const cfg = await this.parseCypressConfig()
+    const pluginsPath = this.getPluginRelativePath(cfg)
+    const blah = Object.entries(cfg).reduce((acc, [key, val]) => {
+      if (key === 'e2e' || key === 'component') {
+        acc = { ...acc, [key]: val }
+      } else {
+        acc = { ...acc, global: { ...acc.global, [key]: val } }
+      }
+
+      return acc
+    }, { global: {}, e2e: {}, component: {} })
+
+    return this.createCypressConfigJs(blah, pluginsPath)
+  }
+
+  private getPluginRelativePath (cfg: Partial<Cypress.ConfigOptions>) {
+    const DEFAULT_PLUGIN_PATH = path.normalize('/cypress/plugins/index.js')
+
+    return cfg.pluginsFile ? cfg.pluginsFile : DEFAULT_PLUGIN_PATH
+  }
+
+  private formatObjectForConfig (obj: Record<string, unknown>, spaces: number) {
+    return JSON.stringify(obj, null, spaces)
+    .replace(/"([^"]+)":/g, '$1:') // remove quotes from fields
+    .replace(/{|(})$/g, '') // remove opening and closing {}
+    .replace(/"/g, '\'') // single quotes
+    .trim()
+  }
+
+  async createCypressConfigJs (config: ConfigOptions, pluginPath: string) {
+    return `const { defineConfig } = require('cypress')
+
+module.export = defineConfig({
+  ${this.formatObjectForConfig(config.global, 2)},
+  e2e: {
+    setupNodeEvents(on, config) {
+      require('${pluginPath}')
+    },
+    ${this.formatObjectForConfig(config.e2e, 4)},
+  },
+  component: {
+    setupNodeEvents(on, config) {
+      require('${pluginPath}')
+    },
+    ${this.formatObjectForConfig(config.component, 4)},
+  },
```

Comment:
This file seems to be the main complexity, I think it's worth adding a test in some capacity - even a simple data context one that does a snapshot, eg given X `cypress.json` it should be converted to Y `cypress.config.js`. What do you think?

Reply:
I was doing this with the e2e test but that's because I thought we weren't doing any unit tests in the in data-context. I'll switch it over to a unit test because it's easier to test different cases there

Reply:
I don't mind how it gets tested, just think it would be good to have something - right now there is no guarantee the generated code is actually a valid JS module (unless I'm missing something). We don't want to get a situation where our migration tool produces invalid code.